api/stacks/deployments/deployer_remote.go

```diff
@@ -0,0 +1,282 @@
+	if err != nil {
+		return errors.WithMessage(err, "unable to create docker client")
+	}
+	defer cli.Close()
+
+	image := getUnpackerImage()
+
+	reader, err := cli.ImagePull(ctx, image, types.ImagePullOptions{})
+	if err != nil {
+		return errors.Wrap(err, "unable to pull unpacker image")
+	}
+	defer reader.Close()
+	io.Copy(io.Discard, reader)
+
+	info, err := cli.Info(ctx)
+	if err != nil {
+		return errors.Wrap(err, "unable to get agent info")
+	}
+	targetSocketBind := getTargetSocketBind(info.OSType)
+
+	composeDestination := filesystem.JoinPaths(stack.ProjectPath, composePathPrefix)
+
+	opts.composeDestination = composeDestination
+
+	cmd := d.buildUnpackerCmdForStack(stack, opts)
+
+	log.Debug().
+		Str("image", image).
+		Str("cmd", strings.Join(cmd, " ")).
+		Msg("running unpacker")
+
+	rand.Seed(time.Now().UnixNano())
+	resp, err := cli.ContainerCreate(ctx, &container.Config{
+		Image: image,
+		Cmd:   cmd,
+	}, &container.HostConfig{
+		Binds: []string{
+			fmt.Sprintf("%s:%s", composeDestination, composeDestination),
+			fmt.Sprintf("%s:%s", targetSocketBind, targetSocketBind),
+		},
+	}, nil, nil, fmt.Sprintf("portainer-unpacker-%d-%s-%d", stack.ID, stack.Name, rand.Intn(100)))
+
+	defer cli.ContainerRemove(ctx, resp.ID, types.ContainerRemoveOptions{})
+
+	if err != nil {
+		return errors.Wrap(err, "unable to create unpacker container")
+	}
+
+	if err := cli.ContainerStart(ctx, resp.ID, types.ContainerStartOptions{}); err != nil {
+		return errors.Wrap(err, "start unpacker container error")
+	}
+
+	statusCh, errCh := cli.ContainerWait(ctx, resp.ID, container.WaitConditionNotRunning)
+	select {
+	case err := <-errCh:
+		if err != nil {
+			return errors.Wrap(err, "An error occurred while waiting for the deployment of the stack.")
+		}
+	case <-statusCh:
+	}
+
+	out, err := cli.ContainerLogs(ctx, resp.ID, types.ContainerLogsOptions{ShowStdout: true, ShowStderr: true})
+	if err != nil {
+		log.Error().Err(err).Msg("unable to get logs from unpacker container")
+	} else {
+		outputBytes, err := io.ReadAll(out)
+		if err != nil {
+			log.Error().Err(err).Msg("unable to parse logs from unpacker container")
+		} else {
+			log.Info().
+				Str("output", string(outputBytes)).
+				Msg("Stack deployment output")
+		}
+	}
+
+	status, err := cli.ContainerInspect(ctx, resp.ID)
+	if err != nil {
+		return errors.Wrap(err, "fetch container information error")
+	}
+
+	if status.State.ExitCode != 0 {
+		return fmt.Errorf("an error occurred while running unpacker container with exit code %d", status.State.ExitCode)
+	}
+
+	return nil
+}
+
+// Creates a docker client with 1 hour timeout
+func (d *stackDeployer) createDockerClient(ctx context.Context, endpoint *portainer.Endpoint) (*dockerclient.Client, error) {
+	timeout := 3600 * time.Second
+	cli, err := d.ClientFactory.CreateClient(endpoint, "", &timeout)
+	if err != nil {
+		return nil, errors.Wrap(err, "unable to create Docker client")
+	}
+
+	// only for swarm
```

Comment:
I think the comments are not synced with the code. I think this is not needed, and one L238 we should state `if not swarm return cli`, L238 can be moved to after the if